src/bots/userLpSettler.ts

```diff
@@ -0,0 +1,383 @@
+				`User for ${this.driftClient.wallet.publicKey.toString()} does not exist`
+			);
+		}
+		this.lookupTableAccount =
+			await this.driftClient.fetchMarketLookupTableAccount();
+
+		const perpMarkets = this.driftClient
+			.getPerpMarketAccounts()
+			.map((m) => m.pubkey);
+
+		logger.info(
+			`Lp settler looking at ${perpMarkets.length} perp markets to determine priority fee`
+		);
+
+		this.priorityFeeSubscriber = new PriorityFeeSubscriber({
+			connection: this.driftClient.connection,
+			frequencyMs: PRIORITY_FEE_SUBSCRIBER_FREQ_MS,
+			addresses: [...perpMarkets],
+		});
+		await this.priorityFeeSubscriber.subscribe();
+		await sleepMs(PRIORITY_FEE_SUBSCRIBER_FREQ_MS);
+
+		// logger.info(`Initializing UserMap`);
+		// const startUserMapSub = Date.now();
+		// await this.userMap.subscribe();
+		// logger.info(`UserMap init took: ${Date.now() - startUserMapSub} ms`);
+
+		logger.info(`${this.name} init'd! took ${Date.now() - start}`);
+	}
+
+	public async reset() {
+		for (const intervalId of this.intervalIds) {
+			clearInterval(intervalId as NodeJS.Timeout);
+		}
+		this.intervalIds = [];
+
+		await this.userMap?.unsubscribe();
+	}
+
+	public async startIntervalLoop(intervalMs?: number): Promise<void> {
+		logger.info(`${this.name} Bot started!`);
+		if (this.runOnce) {
+			await this.trySettleLps();
+		} else {
+			const intervalId = setInterval(this.trySettleLps.bind(this), intervalMs);
+			this.intervalIds.push(intervalId);
+		}
+	}
+
+	public async healthCheck(): Promise<boolean> {
+		let healthy = false;
+		await this.watchdogTimerMutex.runExclusive(async () => {
+			healthy =
+				this.watchdogTimerLastPatTime > Date.now() - 2 * this.defaultIntervalMs;
+		});
+		return healthy;
+	}
+
+	private async trySettleLps() {
+		const start = Date.now();
+		try {
+			const lpsPerMarket: { [key: number]: number } = {};
+			const settleLpIxs: Array<TransactionInstruction> = [];
+
+			logger.info(`Loading users that have been LPs...`);
+			const fetchLpUsersStart = Date.now();
+			const users = await this.fetchLpUsers();
+			logger.info(`Fetch LPs took ${Date.now() - fetchLpUsersStart}`);
+			let usersDoneCount = 0;
+
+			// logger.info(`Going through ${this.userMap!.size()} users...`);
+			// for (const user of this.userMap!.values()) {
+			logger.info(`Going through ${users.length} users...`);
+			for (const userData of users) {
+				const user = userData.userAccount;
+				usersDoneCount++;
+				if (usersDoneCount % 100 === 0) {
+					logger.info(
+						`Processed ${usersDoneCount}/${this.userMap.size()} users...`
+					);
+				}
+				// for (const pos of user.getActivePerpPositions()) {
+				for (const pos of user.perpPositions) {
+					if (pos.lpShares.eq(ZERO)) {
+						continue;
+					}
+
+					if (lpsPerMarket[pos.marketIndex] === undefined) {
+						lpsPerMarket[pos.marketIndex] = 0;
+					} else {
+						lpsPerMarket[pos.marketIndex] += 1;
+					}
+
+					settleLpIxs.push(
+						await this.driftClient.settleLPIx(
```

Comment:
was thinking the rules for settling would be:
1) if user is past initial leverage, settle
2) if it's been longer than an hour since they've been settled, settle
3) if it's 10 min until hour, settle